### ISSUE TITLE
Having problems with making steamauth work with my pretty url template cms.

### ISSUE THREAD

**Issue by Cascoid** [SmItH197/SteamAuthentication#131]
I have been working on a templated cms based on several others I have used previously. I came to have to create the steam login system and my only option was to use this system. The only problem is, it conflicts with my pretty URL's. If I take out pretty URL's, it has problems with finding different pages, but the login system works. If I keep the pretty URL's, the login system doesn't work what-so-ever and I'm pretty sure it's the fact you're using ?login and $_GET['login'] when my beautifier for my URL's already uses the parameter for ?url and $_GET['url']. I also attempted to keep the pretty url's and making the button into an input. It led me to the login page but when it returns, you return to the selected page but with all the shit after /openid/ on steamcommunity when you first go to log in.

ANY help would be greatly helpful and I'm willing to provide most things asked for.

**Comment by Cascoid** [comment]
Just noticed it also receives no data when I do the button with the posting. Really starting to irritate me because I can't rewrite the entire CMS just because of the login system.

**Comment by RedSparr0w** [comment]
if your "pretty urls" litterally just remove the .php i can easily fix that for you

**Comment by Cascoid** [comment]
Through htaccess it redirects all url requests to index.php which then uses the request to find the correct content for that url request. I have removed the .php's to test it but then it cannot find the files within the template system since the link is looking in the main directory instead of being redirected by index.php.

htaccess:
`RewriteEngine On

RewriteRule ^(|/)$ index.php?url=$1
RewriteRule ^([a-zA-Z0-9_-]+)(|/)$ index.php?url=$1`

**Comment by RedSparr0w** [comment]
could you instead use something like this for your pretty urls?
```
RewriteEngine On
RewriteCond %{REQUEST_FILENAME} !-f 
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^([A-Za-z0-9\-\_]+)/$ $1.php [L]
RewriteRule ^([A-Za-z0-9\-\_]+)$ $1.php [L]
```
which does: example.com/thispage --> example.com/thispage.php
which it seems like you may already be doing something similar inside your index.php file with the url param

as im fairly sure it is the fact you have the ?url in the address, as it cant be sent as a redirect from steam due to steam escaping the chars

**Comment by Cascoid** [comment]
Exactly what happened earlier. Works for index.php since it's in the main directory and the main reference file for the CMS, but as soon as you call for another page it doesn't look for it. Here's the layout of my CMS with the main things you need to know.
```
APPLICATION // main folder for most of cms
  MANAGEMENT
    config.php // management for the entire cms apart from steamauth
  CLASSES
    engine.php // mysql stuff
    core.php // checks if maintenance is enabled, error exceptions etc
    template.php // used to structure the template using the main file (index.php) and the contents of the template
    output.php // reads every javascript and css file, outputs it in html form for the template.
  TEMPLATE
    MAINTEMPLATE
      index.php // just the front page content
      support.php // support page content
      404.php // 404 content
      maintenance.php // maintenance page content
index.php // main reference point for each file and file structure. It outputs the same thing on every page and is used to output every page using htaccess.
global.php // brings the whole cms together to work 
```
What the website is doing when it's using my htaccess is it's saying "Oh, he wants to go to support. We'll ask for index.php to search for the contents of support.php and return it to the user as /support."

This is likely where the problem lies because with your method (similar to one I tried earlier), it stops my CMS from being able to reach it's file because it needs the ?url= (unless there's an alternative way, that is).

**Comment by RedSparr0w** [comment]
any chance you could turn on php error reporting and see what errors it throws out? also are you including steamauth.php at the start of your index.php?

**Comment by Cascoid** [comment]
Will do that now. It's at the very top, yes.

**Comment by Cascoid** [comment]
Giving me no errors what-so-ever. Just saying there is no file called support.php since it's searching for it and it's not in the main directory.

**Comment by RedSparr0w** [comment]
just to be clear you are including steamauth.php in the home index.php file not the template/maintemplate/index.php file?

**Comment by Cascoid** [comment]
Yes. I tried it out of that file and in the individual file instead and nothing changed. Worked exactly the same since it just makes a template using the initial file.

**Comment by RedSparr0w** [comment]
any chance you could post a link to your website or [email me](mailto:admin@redsparr0w.com) one?

only other thing i can think of right now is setting a loginpage in the SteamConfig.php file

**Comment by Cascoid** [comment]
Unfotunately I have not got it on a live site currently.

**Comment by RedSparr0w** [comment]
one thing that may work is if you try going to example.com?login

**Comment by Cascoid** [comment]
Didn't fix through that, unfortunately. Starting to look like the only option is the button method but that returns a ridiculously long url and doesn't actually sign you in.

**Comment by Cascoid** [comment]
Found a horrible way of making it work but it will have to do unless anyone else can suggest anything. I basically used your htaccess and then added a line for the error document 404. I made a file called 404.php in the main directory. I then made a file for each file that is in my template and made all of these files and the 404.php require_once the main index.php file. Really not the cleanest way to do it at all but if no-one else knows anything different I guess I'll have to stick to it.
